lib/presentation/widgets/m3_widgets/buttons/google_sing_in_button.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:flutter/material.dart';
+import 'package:trip_tally/presentation/theme/app_colors.dart';
+import 'package:trip_tally/presentation/theme/app_dimensions.dart';
+import 'package:trip_tally/presentation/theme/app_paths.dart';
+import 'package:trip_tally/presentation/utils/enums/context_extensions.dart';
+
+class GoogleSignInButton extends StatelessWidget {
+  const GoogleSignInButton({required this.onPressed, super.key});
+
+  final VoidCallback onPressed;
+
+  @override
+  Widget build(BuildContext context) {
+    return ElevatedButton(
+      onPressed: onPressed,
+      style: ElevatedButton.styleFrom(
+        backgroundColor: AppColors.white,
+        foregroundColor: AppColors.black,
+        padding: const EdgeInsets.symmetric(
+          horizontal: AppDimensions.d16,
+          vertical: AppDimensions.d12,
+        ),
+        shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(AppDimensions.d8),
+        ),
+        elevation: AppDimensions.d3,
+      ),
+      child: Row(
+        children: [
+          Image.asset(
```

Comment:
meybe, it would be better to extract the icon into separate widget. we have  apple icon and google icon.